File: TestApplication/TestApplication/AccountOperations.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TestApplication
{
    public class AccountOperations
    {
        private IAccountData accountData;

        public AccountOperations(IAccountData accountData) 
        {
            this.accountData = accountData ?? throw new ArgumentNullException(nameof(accountData));
        }

        public int GetBalanceForSingleAccount(int accountId) 
        {
            int returnBalance = 0;

            IAccount account = accountData.GetAccount(accountId);
            if (account != null)
            {
                returnBalance = account.Balance;
            }

            return returnBalance;
        }
    }
}

File: TestApplication-Java/src/main/java/com/scottgarryfoster/AccountOperations.java
package com.scottgarryfoster;

import com.scottgarryfoster.API.IAccountData;

public class AccountOperations
{
    public AccountOperations(IAccountData accountData)
    {
        if (accountData == null)
        {
            throw new NullPointerException(IAccountData.class + " may not be null.");
        }
    }
}
